src/main/java/seedu/address/commons/core/index/IndexComparator.java

```diff
@@ -0,0 +1,13 @@
+package seedu.address.commons.core.index;
+
+import java.util.Comparator;
+
+/**
+ * Represents a comparator for sorting indices in descending order.
+ */
+public class IndexComparator implements Comparator<Index> {
```

Comment:
Nit: would it be good to rename this IndexReverseComparator or something along those lines? Just so it's clear from the class name that this sorts in descending order.